IdentityCore/src/cache/accessor/MSIDAccountCredentialCache.m

```diff
@@ -348,11 +352,7 @@ - (BOOL)removeAccount:(nonnull MSIDAccountCacheItem *)account
 
     MSID_LOG_WITH_CTX_PII(MSIDLogLevelVerbose, context, @"(Default cache) Removing account with environment %@, user ID %@, username %@", account.environment, MSID_PII_LOG_MASKABLE(account.homeAccountId), MSID_PII_LOG_EMAIL(account.username));
 
-    MSIDDefaultAccountCacheKey *key = [[MSIDDefaultAccountCacheKey alloc] initWithHomeAccountId:account.homeAccountId
-                                                                                   environment:account.environment
-                                                                                         realm:account.realm
-                                                                                          type:account.accountType];
-
+    MSIDCacheKey *key = [account generateCacheKey];
```

Comment:
note: this one wasn't setting username before because it wouldn't remove an item if its username changed.... Should we keep the same?

Reply:
Please see comment above.